### PR TITLE
feat: calendar newest styles adaptation

### PR DESCRIPTION
**Pull request by JKMarkowski**
#### Please provide a link to the associated issue.
fixes: https://github.com/SAP/fundamental-ngx/issues/2130
#### Please provide a brief summary of this pull request.
In this PR there are newest styles adaptation changes, including new options and ways to customization. It's worth to say that there is also new view for aggregated year. 
#### New Properties: 
```html
[compact]
[markWeekends]
[specialDaysRules]
[showWeekNumbers]
[aggregatedYearGrid]
[yearGrid]
[mobilePortrait]
[mobileLandscape]
```

#### Code:
Calendar service now is more complex. It provides functionality for keyboard support for all views, not only year/month as it was before. Those changes also remove `setTimeout`, which caused some issues/bugs. There are included some new data models and interfaces.

#### Removed:
[blockFunction] and range block functions: There are only disable function left.

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

**Comment by mikerodonnell89** [comment]
For the mobile dialog modes - the specs state there should be some side margins, coming from the dialog

File: apps/docs/src/app/core/component-docs/calendar/calendar-docs.component.ts [hunk 1]
```diff
@@ -63,6 +69,42 @@ myDisableFunction = function(d: FdDate): boolean {
     return d.getTimeStamp() > firstDay.getTimeStamp() && d.getTimeStamp() < lastDay.getTimeStamp()
 }`;
 
+
+    exampleSpecialDays = `Example Special Days Functions:
+
+// Mark weekends with special day number 10
```
Comment (mikerodonnell89) [inline apps/docs/src/app/core/component-docs/calendar/calendar-docs.component.ts:75]: This section of the docs is a little confusing, I thought that the example would use this rule.  I think the better option would be to use the more complex rule at the bottom in the example

**Comment by mikerodonnell89** [comment]
Compact calendar looks different here: 
Screen Shot 2020-04-07 at 12 24 32 PM

compared to styles: 
Screen Shot 2020-04-07 at 12 24 37 PM

notice the styles days are squares and the ngx ones are rectangular

File: apps/docs/src/app/core/component-docs/calendar/calendar-docs.component.html [hunk 2]
```diff
@@ -22,6 +21,50 @@
...
+<description>
+    Year Grid and Aggregated Year Grid can be customized, by passing <code>[yearGrid]</code> and <code>[aggregatedYearGrid]</code>
+    properties. The columns and rows amount can be modified, same as the way how year is displayed.
+</description>
+<component-example>
+    <fd-calendar-grid-example></fd-calendar-grid-example>
+</component-example>
+<code-example [exampleFiles]="calendarGridSource"></code-example>
+
+<separator></separator>
```
Comment (InnaAtanasova) [inline apps/docs/src/app/core/component-docs/calendar/calendar-docs.component.html:66]: @Vanessa-Cusmich could you take a look at the doc part? Thanks

**Comment by JKMarkowski** [comment]
> For the mobile dialog modes - the specs state there should be some side margins, coming from the dialog

Done

File: apps/docs/src/app/core/component-docs/calendar/calendar-docs.component.ts (hunk 1 above)
Comment (JKMarkowski) [reply apps/docs/src/app/core/component-docs/calendar/calendar-docs.component.ts:77]: Done, take a look at new section

**Comment by stefanoScalzo** [comment]
What if the user had the ability to choose how many years to skip by.
Also not sure if ui5 isn't implementing their click on their year-selector properly or if it is supposed to have a grey highlight when clicked
Screen Shot 2020-04-08 at 9 13 51 PM
Screen Shot 2020-04-08 at 9 13 41 PM

**Comment by JKMarkowski** [comment]
> What if the user had the ability to choose how many years to skip by.

It's done under the hood, it calculates how much years are displayed at once and switch by this amount

File: libs/core/src/lib/calendar/calendar.component.ts [hunk 3]
```diff
@@ -404,6 +468,12 @@ export class CalendarComponent implements OnInit, ControlValueAccessor, Validato
         this.dayViewComponent.focusActiveDay();
     }
 
+    public selectedYears(yearsSelected: AggregatedYear) {
```
[the remark this answers is not in the document]
Comment (JKMarkowski) [reply libs/core/src/lib/calendar/calendar.component.ts]: Done

File: libs/core/src/lib/calendar/calendar-views/calendar-year-view/calendar-year-view.component.ts [hunk 4]
```diff
@@ -73,24 +77,18 @@ export class CalendarYearViewComponent implements AfterViewChecked, OnInit, OnDe
...
-    ngAfterViewChecked(): void {
-        if (this.newFocusedYearId) {
-            this.focusElement(this.newFocusedYearId);
-            this.newFocusedYearId = null;
-        }
-    }
-
     /** @hidden */
     ngOnInit(): void {
+        this.calendarService.colAmount = this.yearViewGrid.cols;
```
[the remark this answers is not in the document]
Comment (JKMarkowski) [reply libs/core/src/lib/calendar/calendar-views/calendar-year-view/calendar-year-view.component.ts]: Done

[the remark this answers is not in the document]
Comment (JKMarkowski) [reply libs/core/src/lib/calendar/calendar-views/calendar-year-view/calendar-year-view.component.ts]: Done

File: libs/core/src/lib/calendar/calendar-views/calendar-aggregated-year-view/calendar-aggregated-year-view.component.ts [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+    selectYear(selectedYear: AggregatedYear, event?: MouseEvent) {
+        if (event) {
+            event.stopPropagation();
+        }
+        this.yearsSelected = selectedYear;
+        this.yearsClicked.emit(this.yearsSelected);
+    }
+
+    /** Method for handling the keyboard navigation. */
+    onKeydownYearHandler(event, index: number): void {
```
[the remark this answers is not in the document]
Comment (JKMarkowski) [reply libs/core/src/lib/calendar/calendar-views/calendar-aggregated-year-view/calendar-aggregated-year-view.component.ts:104]: Done

File: libs/core/src/lib/calendar/calendar-views/calendar-aggregated-year-view/calendar-aggregated-year-view.component.ts [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+
+
+    /** @hidden */
+    ngOnDestroy(): void {
+        this.onDestroy$.next();
+        this.onDestroy$.complete();
+    }
+
+    /** Method that sends the year to the parent component when it is clicked. */
+    selectYear(selectedYear: AggregatedYear, event?: MouseEvent) {
```
[the remark this answers is not in the document]
Comment (JKMarkowski) [reply libs/core/src/lib/calendar/calendar-views/calendar-aggregated-year-view/calendar-aggregated-year-view.component.ts:95]: Done

File: libs/core/src/lib/calendar/calendar-views/calendar-aggregated-year-view/calendar-aggregated-year-view.component.ts [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+     */
+    @Input()
+    yearViewGrid: CalendarYearGrid;
+
+    /** Event fired when a year is selected. */
+    @Output()
+    readonly yearsClicked: EventEmitter<AggregatedYear> = new EventEmitter<AggregatedYear>();
+
+    /** Parameter storing first shown year on list */
+    private firstYearInList: number;
```
[the remark this answers is not in the document]
Comment (JKMarkowski) [reply libs/core/src/lib/calendar/calendar-views/calendar-aggregated-year-view/calendar-aggregated-year-view.component.ts:64]: Done

File: libs/core/src/lib/calendar/calendar-views/calendar-aggregated-year-view/calendar-aggregated-year-view.component.spec.ts [hunk 8]
```diff
@@ -0,0 +1,72 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { CalendarAggregatedYearViewComponent } from './calendar-aggregated-year-view.component';
+import { CalendarService } from '../../calendar.service';
+import { AggregatedYear } from '@fundamental-ngx/core';
```
[the remark this answers is not in the document]
Comment (JKMarkowski) [reply libs/core/src/lib/calendar/calendar-views/calendar-aggregated-year-view/calendar-aggregated-year-view.component.spec.ts]: Done

File: libs/core/src/lib/calendar/calendar-views/calendar-aggregated-year-view/calendar-aggregated-year-view.component.html [hunk 9]
```diff
@@ -0,0 +1,25 @@
+<div class="fd-calendar__years">
+    <table class="fd-calendar__table" role="grid">
+        <tbody class="fd-calendar__group">
+        <tr class="fd-calendar__row"
+            *ngFor="let colYears of calendarYearListGrid; let rowIndex = index">
+            <td class="fd-calendar__item"
```
[the remark this answers is not in the document]
Comment (JKMarkowski) [reply libs/core/src/lib/calendar/calendar-views/calendar-aggregated-year-view/calendar-aggregated-year-view.component.html:6]: Done

File: apps/docs/src/app/documentation/shared-core-modules.ts [hunk 10]
```diff
@@ -15,6 +15,7 @@ export const sharedCoreModules = [
     ButtonModule,
     InputGroupModule,
     ShellbarModule,
+    DialogModule,
```
[the remark this answers is not in the document]
Comment (JKMarkowski) [reply apps/docs/src/app/documentation/shared-core-modules.ts]: Changed only to be in calendar docs module

File: libs/core/src/lib/calendar/calendar-header/calendar-header.component.html [hunk 11]
```diff
@@ -1,10 +1,9 @@
 <header class="fd-calendar__header" aria-live="polite">
-    <div class="fd-calendar__navigation">
-        <div class="fd-calendar__action">
+    <div class="fd-calendar__navigation fd-calendar__navigation--main">
```
[the remark this answers is not in the document]
Comment (JKMarkowski) [reply libs/core/src/lib/calendar/calendar-header/calendar-header.component.html]: `--main` added only on landscape

File: libs/core/src/lib/calendar/calendar-views/calendar-aggregated-year-view/calendar-aggregated-year-view.component.ts [hunk 12]
```diff
@@ -0,0 +1,244 @@
...
+    constructor(
+        private eRef: ElementRef,
+        private changeDetectorRef: ChangeDetectorRef,
+        private calendarService: CalendarService) {
+    }
+
+
+    /** @hidden */
+    ngOnInit(): void {
+        this.calendarService.rowAmount = this.aggregatedYearsViewGrid.rows;
```
[the remark this answers is not in the document]
Comment (JKMarkowski) [reply libs/core/src/lib/calendar/calendar-views/calendar-aggregated-year-view/calendar-aggregated-year-view.component.ts]: Done

**Pull request merged**